comit/src/expiries.rs

```diff
@@ -270,13 +318,96 @@ fn period_for_bob_to_complete(config: &Config, current_state: BobState) -> Durat
     period_to_complete(config, current_state, Duration::zero())
 }
 
+// Scale the duration by factor / 100 i.e., scale the duration by a
+// percentage.
+#[allow(clippy::cast_precision_loss, clippy::cast_possible_truncation)]
+fn scale_by_factor(d: Duration, factor: u32) -> Duration {
+    let secs = d.whole_seconds();
+    let scaled = secs * factor as i64;
+    let reduced = (scaled as f64 / 100.0).floor();
+
+    Duration::seconds(reduced as i64)
+}
+
+// Its super easy to mix these up, add types so the compiler saves us.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct AlphaOffset(Duration);
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct BetaOffset(Duration);
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct AlphaExpiry(Timestamp);
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct BetaExpiry(Timestamp);
+
+macro_rules! impl_from {
+    ($from:tt, $target:tt) => {
+        impl From<$from> for $target {
+            fn from(f: $from) -> Self {
+                f.0
+            }
+        }
+    };
+}
+impl_from!(AlphaOffset, Duration);
+impl_from!(BetaOffset, Duration);
+impl_from!(AlphaExpiry, Timestamp);
+impl_from!(BetaExpiry, Timestamp);
+
+macro_rules! impl_from_nested {
+    ($from:tt, $target:tt) => {
+        impl From<$from> for $target {
+            fn from(f: $from) -> Self {
+                Self(f)
+            }
+        }
+    };
+}
+impl_from_nested!(Duration, AlphaOffset);
+impl_from_nested!(Duration, BetaOffset);
+impl_from_nested!(Timestamp, AlphaExpiry);
+impl_from_nested!(Timestamp, BetaExpiry);
+
+impl fmt::Display for AlphaOffset {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let secs = self.0.whole_seconds();
+        write!(f, "{}", human_readable(secs))
+    }
+}
+
+impl fmt::Display for BetaOffset {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let secs = self.0.whole_seconds();
+        write!(f, "{}", human_readable(secs))
+    }
+}
+
+fn human_readable(seconds: i64) -> String {
```

Comment:
This looks like a `Display` implementation.

Reply:
This is a helper function, called from the `Display` imlps.